src/Microsoft.AspNetCore.Server.Kestrel/Filter/StreamSocketOutput.cs

```
@@ -65,14 +70,39 @@ public void ProducingComplete(MemoryPoolIterator end)
             var block = _producingBlock;
             while (block != end.Block)
             {
-                _outputStream.Write(block.Data.Array, block.Data.Offset, block.Data.Count);
+                // Writes can fail if the client has closed the connection but we're still sending a response.
+                // If we don't handle an exception from _outputStream.Write() here, we'll leak memory blocks.
+                if (_canWrite)
+                {
+                    try
+                    {
+                         _outputStream.Write(block.Data.Array, block.Data.Offset, block.Data.Count);
+                    }
+                    catch (Exception e)
```

This exception is lost